cpp/src/quantiles/quantiles.cu

```diff
@@ -16,196 +16,253 @@
 
 //Quantile (percentile) functionality
 
-#include "quantiles.h"
+#include "quantiles.hpp"
 #include "utilities/cudf_utils.h"
 #include "utilities/error_utils.hpp"
 #include "utilities/type_dispatcher.hpp"
+#include "utilities/wrapper_types.hpp"
 #include "rmm/thrust_rmm_allocator.h"
 #include "cudf.h"
 
 #include <thrust/device_vector.h>
 #include <thrust/copy.h>
 
-namespace{ //unknown
-  template<typename VType,
-           typename RetT = double>
-    void f_quantile_tester(rmm::device_vector<VType>& d_in)
+
+namespace{ // anonymouys namespace
+
+  struct QuantiledIndex {
+    gdf_size_type lower_bound;
+    gdf_size_type upper_bound;
+    gdf_size_type nearest;
+    double fraction;
+  };
+
+  QuantiledIndex find_quantile_index(gdf_size_type length, double quant)

```

Comment:
Hmmm, could this function just be the constructor for `QuantileIndex`? e.g. `QuantileIndex(gdf_size_type length, double quant);`

Reply:
Good pointing! yes it is the constructor.
I will do it.